Piwik.Tracker/PiwikTracker.cs

```diff
@@ -1160,14 +1159,15 @@ public void SetIp(string ip)
         /// <exception cref="Exception"/>
         public void SetUserId(string userId)
         {
+            // todo this is contradicion: (userId == null) / string.IsNullOrEmpty(userId)

```

Comment:
Thanks for clearifiying this, you are right. Maybe we make the argument userId optional: 
`public void SetUserId(string userId = null)`